src/components/application_manager/rpc_plugins/vehicle_info_plugin/src/vehicle_info_plugin.cc

```diff
@@ -158,7 +173,33 @@ void VehicleInfoPlugin::DeleteSubscriptions(
     }
   }
 }
+
+void VehicleInfoPlugin::SelectVehicleDataForUnsubscribe(
+    const app_mngr::ApplicationSharedPtr& app,
+    std::set<mobile_apis::VehicleDataType::eType>& vehicle_data_for_unscribe) {
+  VehicleInfoAppExtension& vehicle_extension =
+      VehicleInfoAppExtension::ExtractVIExtension(*app);
+  vehicle_data_for_unscribe = vehicle_extension.Subscriptions();
+  application_manager::ApplicationSet applications =
+      application_manager_->applications().GetData();
+  if (!applications.empty()) {
```

Comment:
@IGapchuk I would say that it is redundant check, or move it before using unsubscriber

Reply:
This check needs for not doing redundant selecting vehicle data different with other applications. Because, if we haven't other applications, that vehicle data of specified application will be only one data for unsubscribe.